eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/io/FastaSynonymSequenceReader.java

```diff
@@ -30,6 +30,22 @@ public FastaSynonymSequenceReader(ContigMapping contigMapping, Path fastaPath) t
         this.contigMapping = contigMapping;
     }
 
+    @Override
+    public boolean doesContigExist(String contig) {
+        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(contig);
+
+        if (contigSynonyms == null) {
+            throw new IllegalArgumentException(
+                    "Contig '" + contig + "' not found in the assembly report");
+        }
+
+        return super.doesContigExist(contigSynonyms.getSequenceName())
+                || super.doesContigExist(contigSynonyms.getGenBank())
+                || super.doesContigExist(contigSynonyms.getRefSeq())
+                || super.doesContigExist(contigSynonyms.getUcsc())
+                || super.doesContigExist(contigSynonyms.getAssignedMolecule());
```

Comment:
The issue you mention won't be a problem because `ContigMapping` will set the assigned molecule to null if the sequence-role is not "assembled-molecule". So only one row (the correct one) will match `1`.

however, I added the condition to check for synonyms only if genbank and refseq are identical, to match the logic in `getSequence`.

There's an issue pending, where a given assigned-molecule appears several times as "assembled-molecule", but I plan to make that change in the eva-accession-import branch 0.2.x and then forward-port those changes into master